### ISSUE TITLE
react-native-intercom not working on iOS13

### ISSUE THREAD

**Issue by Miriloper** [tinycreative/react-native-intercom#320]
Hi!

I've integrated Intercom on my app, is working on Android but it crash on iOS (i've checked different versions, and happens the same). 

I am not using pods right now, all is installed manually.

```
info === BUILD TARGET RNIntercom OF PROJECT RNIntercom WITH CONFIGURATION Debug ===

info
Check dependencies

info Argument list too long: recursive header expansion failed at /Users/myuser/projects/this-project/node_modules/react-native-intercom/iOS/../../../ios/build/myproject/Index/DataStore/v5/records/2F.

info

error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening myproject.xcworkspace

** BUILD FAILED **


The following commands produced analyzer issues:
	Analyze /Users/myuser/projects/this-project/node_modules/react-native/ReactCommon/jsi/jsi.cpp normal x86_64
	Analyze /Users/myuser/projects/this-project/node_modules/react-native/ReactCommon/yoga/yoga/Yoga.cpp normal x86_64
	Analyze Base/RCTModuleMethod.mm normal x86_64
(3 commands with analyzer issues)

The following build commands failed:
	Check dependencies
(1 failure)
. Run CLI with --verbose flag for more details.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! myproject@0.1.0 ios: `ENVFILE=.env.dev react-native run-ios --scheme myproject-dev`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the myproject@0.1.0 ios script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/myuser/.npm/_logs/2019-10-14T16_01_03_555Z-debug.log
```

Any idea how to fix it? Thanks!

**Comment by smakosh** [comment]
Same here but I'm using the pod

```
pod 'react-native-intercom', :path => '../node_modules/react-native-intercom'
```

**Comment by bogdan-vol** [comment]
@Miriloper It works for me, at least on an emulator with iOS 13, but I have everything installed through cocoa pods. React native version is 0.60


**package.json:** `{
  ...,
  "dependencies": {
    ...,
    "react": "16.6.1",
    "react-native": "^0.60.0",
    "react-native-intercom": "^13.1.0"
  },
  ...
}`


**Podfile:** `target 'Drumeo' do
  rn_path = '../node_modules/react-native'
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/React'
  pod 'React-DevSupport', :path => '../node_modules/react-native/React'
  pod 'React-fishhook', :path => '../node_modules/react-native/Libraries/fishhook'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-RCTWebSocket', :path => '../node_modules/react-native/Libraries/WebSocket'
  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'react-native-intercom', :path => '../node_modules/react-native-intercom'
  ...
end
post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == "React"
      target.remove_from_project
    end
    targets_to_ignore = %w(React yoga)
    if targets_to_ignore.include? target.name
      target.remove_from_project
    end
  end
end`


**react-native-intercom.podspec:** `require 'json'
package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
Pod::Spec.new do |s|
  s.name         = package['name']
  s.version      = package['version'].sub('-beta', '')
  s.license      = { :type => 'MIT' }
  s.homepage     = package['homepage']
  s.authors      = package['contributors'].flat_map { |author| { author['name'] => author['email'] } }
  s.summary      = package['description']
  s.source       = { :git => package['repository']['url'] }
  s.source_files = 'iOS/*.{h,m}'
  s.platform     = :ios, '8.0'
  s.frameworks   = [ "Intercom" ]
  s.static_framework = true
  s.dependency 'React'
  s.dependency 'Intercom', '~> 5.0'
end`

**Comment by vgm8** [comment]
I had this exact same error and I managed to solve. I started in a fresh branch doing `npm install --save react-native-intercom` and `react-native link react-native-intercom`.

Then in my Podfile I replace `pod 'react-native-intercom', :path => '../node_modules/react-native-intercom'`  for `pod 'Intercom', '~> 5.4.1'` and install pods.  I use 5.4.1 because the last version doesn't work on Xcode 10.

Go to Xcode > [YOUR_PROYECT] > Libraries and add manually all these files from react-native-intercom

![image](https://user-images.githubusercontent.com/25029191/70215970-ed028d00-173e-11ea-97b3-35f07d48c023.png)

I think we can close this issue. I've used this in two of my projects and both of them work perfect. Hope it helps.
